### PR TITLE
Added epoch_md5 document ID type

### PR DESCRIPTION
**Pull request by cdahlqvist**
Added a new document ID type that is made up of an epoch prefixed MD5 hash, and is a bit shorter and more efficient than the epoch prefixed UUID. I also increased the size of the shard generated in the corresponding challenge to better see how index throughput varies with shard size.

### CODE REVIEW

**Comment by cdahlqvist** [comment]
@danielmitterdorfer Refactored challenge and corrected README.

File: eventdata/challenges/document_id_benchmark.json [hunk 1]
```diff
@@ -1,350 +1,182 @@
 {% set p_bulk_indexing_clients = (bulk_indexing_clients | default(20)) %}
-{% set p_iterations = bulk_indexing_iterations | default(150000) %}
+{% set p_iterations = bulk_indexing_iterations | default(200000) %}
 {% set p_iterations_per_client = (p_iterations / p_bulk_indexing_clients) | int %}
 
 {
   "name": "document_id_evaluation",
-  "description": "Index about 30GB of data into five single shard indices using different document ID types. IDs are autogenerated by Elasticsearch, meaning there are no conflicts.",
+  "description": "Indexes about 40GB of data into seven single shard indices using different document ID types. IDs are autogenerated by Elasticsearch, meaning there are no conflicts.",
```
Comment (danielmitterdorfer) [inline eventdata/challenges/document_id_benchmark.json]: In the README you mentioned disk space requirements. Do they need to be updated as well?

Comment (cdahlqvist) [reply eventdata/challenges/document_id_benchmark.json]: Good catch. :-)

File: eventdata/parameter_sources/metricbeat_bulk_source.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        response = { "body": "\n".join(bulk_array), "action-metadata-present": True, "bulk-size": self._bulk_size }
+
+        if "pipeline" in self._params.keys():
+            response["pipeline"] = self._params["pipeline"]
+
+        return response
+
+    def __get_uuid(self):
+        u = str(uuid.uuid4())
+        return u[0:8] + u[9:13] + u[14:18] + u[19:23] + u[24:36]
```
[the remark this answers is not in the document]
Comment (cdahlqvist) [reply eventdata/parameter_sources/metricbeat_bulk_source.py]: It is indeed...

File: eventdata/parameter_sources/metricbeat_bulk_source.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+    def __get_epoch_uuid(self):
+        u = self.__get_uuid()
+        ts = int(time.time())
+
+        if(self._id_delay_probability > 0 and self._id_delay_probability < random.random()):
+            ts = ts - self._id_delay_secs
+
+        return hex(ts)[2:10] + u
```
[the remark this answers is not in the document]
Comment (cdahlqvist) [reply eventdata/parameter_sources/metricbeat_bulk_source.py]: That is cleaner, so have updated.

**Comment by cdahlqvist** [comment]
@dliappis Have updated. Better now?

**Comment by dliappis** [comment]
Thanks @cdahlqvist for the fixes!

**Pull request merged**